open-craft/frontend-app-library-authoring
# SE-2946: Add support for Video and Problem library types.

Adds support for library types, adding the 'video' and 'problem' type, and permitting filtering by these types.

**Discussions**: https://docs.google.com/document/d/1hRrSmWWlGHhi-bh9AIlpSuLtJFnn2GpavNe8OQxthbE/edit

**Dependencies**: https://github.com/edx/frontend-app-library-authoring/pull/7 must be merged first. This pull requests exists to make code review easier-- it is not the one to be merged.

**Screenshots**:

Creation page changes:

<img width="983" alt="image" src="https://user-images.githubusercontent.com/1099483/91499806-c3b6fd80-e887-11ea-841a-443ed6f3fdd8.png">

Edit Details:

<img width="983" alt="image" src="https://user-images.githubusercontent.com/1099483/91500546-5c01b200-e889-11ea-9447-04d5f29afa8e.png">

Listing with filtering:

<img width="1359" alt="image" src="https://user-images.githubusercontent.com/1099483/91500807-cfa3bf00-e889-11ea-81c1-586221e45f13.png">


**Merge deadline**: This PR not to be merged-- it is only for code review.

**Testing instructions**:

1. Use the `edx-platform` branch `open-craft:library_types` on the backend.
1. Follow the directions in https://github.com/edx/frontend-app-library-authoring/pull/4 up until item 15.
1. Create a new library of each type.
1. Create blocks in the library. In complex and legacy libraries, you should be able to create many different types.
1. In Video and problem libraries, you should be able to only create blocks of their respective type.
1. For complex, video, and problem libraries, test conversion by going to the Details menu for the library and switching the type.
    * This should succeed if the library is empty always.
    * It should succeed if moving from video or problem to complex.
    * It should succeed if moving from complex to problem, or complex to video, when all blocks match their respective types.
    * It should fail if moving from complex to problem, or problem to complex, if all blocks don't match the target type.
1. You should be able to filter the libraries by type on the library listing page. 

**Reviewers**
- [x] @yekibud
